input-profile.ts

```diff
@@ -66,23 +81,23 @@ export class InputProfile extends Component {
                 'inputsourceschange',
                 this.onInputSourcesChange.bind(this)
             );
+            this._registeredFlag = true;
         });
     }
 
     onDeactivate() {
-        this.engine.onXRSessionStart.add(() => {
+        if (this._registeredFlag)
```

Comment:
No need for this flag, if you remove the listener and it doesn't exist, it shouldn't throw an error